src/data/datainterfaces.ts

```diff
@@ -230,6 +530,20 @@ export interface IFieldType {
   isConfigValid(fieldDefinition: IFieldDefinition): { valid: boolean; message?: string };
 }
 
+const IFieldTypeObject: IFieldType = {
+  getType: (Joi.function().arity(0).required() as unknown) as () => FieldType,
+  getName: (Joi.function().arity(0).required() as unknown) as () => string,
+  getInput: (Joi.function().maxArity(6).minArity(5).required() as unknown) as () => JSX.Element,
+  renderValue: (Joi.function().maxArity(5).minArity(3).required() as unknown) as () => JSX.Element,
+  renderValueForEmail: (Joi.function().maxArity(5).minArity(3).required() as unknown) as () => JSX.Element,
+  getSettingsButton: (Joi.function().arity(3).required() as unknown) as () => JSX.Element,
+  isVisible: (Joi.function().arity(0).required() as unknown) as () => boolean,
+  isValid: (Joi.function().arity(2).required() as unknown) as () => boolean,
+  isConfigValid: (Joi.function().arity(1).required() as unknown) as () => { valid: boolean; message?: string },
+};
```

Comment:
Could we test this, too? It feels a little dangerous as we might forget to adjust these numbers similar to the enums.